tests/integration_tests/scheduler/test_generic_driver.py

```diff
@@ -138,9 +140,13 @@ async def started(iens):
         f"sleep {job_kill_window}; touch {tmp_path}/survived",
         name="kill_me",
     )
-    await poll(driver, {0}, started=started)
+    await poll(driver, {0}, started=kill_job_once_started)
 
     # Give the script a chance to finish if it is running
-    await asyncio.sleep(test_grace_time)
+    async def check_if_survived_file_exists():
```

Comment:
I don't see how this improves the test? It will always wait  until the grace period is finished anyhow.

Reply:
The idea was that if the test is failing after 20 seconds; we won't have to wait the full 120s.

Reply:
Okay, will remove 👌